doc/configuration.md

```diff
@@ -0,0 +1,184 @@
+# Configuration
+
+The plugin supports a number of configuration options. These can be supplied when registering the plugin.
+
+## Applying Configurations
+Kong support a wide variety of way to configure the plugin. It can done using [Kong Admin API]( configurations) or opt for the [Declarative Configuration in DB-less mode](https://docs.konghq.com/gateway/latest/production/deployment-topologies/db-less-and-declarative-config/). In Kubernetes environments, numerous configuration methods exist, which are not covered here to keep the documentation concise. However, you can derive these methods based on the provided examples.
+
+## Configuration Precedence
+Configuration can be set either through the plugin configuration field or environment variables, each with its precedence order. Here is the precedence sequence:
+1. Environment variables.
+2. Plugin configuration field.
+3. Default value.
+
+## Configuration Fields
+
+| Name | Environment Variable | Description | Since | Type | Default |
+| ---- | -------------------- | ----------- | ----- | ---- | ------- |
+| `agent_host` | `DD_AGENT_HOST` | Hostname or IP to reach the Datadog Agent | `v0.2.0` | `string` | `localhost` |
+| `trace_agent_port` | `DD_TRACE_AGENT_PORT` | Port to reach the Datadog Agent | `v0.2.0` | `number` | `8126` |
+| `trace_agent_url` | `DD_TRACE_AGENT_URL` | URL used to reach the Datadog Agent | `v0.2.0` | `string` | `http://localhost:8126` |
+| `service_name` | `DD_SERVICE` | Name of the service that is producing traces | `v0.0.1` | `string` | Service registered by Kong or `kong` |
+| `environment` | `DD_ENV` | Add `env` tag for [Unified Service Tagging](https://docs-staging.datadoghq.com/dmehala/cpp-updates/getting_started/tagging/unified_service_tagging/?tab=kubernetes)  | `v0.0.1` | `string` | `nil` |
```

Comment:
👋🏻 links to unified service tagging are incorrect - 404; probably internal link for you only:

- https://docs-staging.datadoghq.com/dmehala/cpp-updates/getting_started/tagging/unified_service_tagging/?tab=kubernetes

vs

- https://docs.datadoghq.com/getting_started/tagging/unified_service_tagging/?tab=kubernetes

Reply:
Addressed in https://github.com/DataDog/kong-plugin-ddtrace/pull/72. Thanks for flagging @Kyslik